### PR TITLE
Example Workflow Docs on How to Create Tests for One's Own Package

### PR DESCRIPTION
**Pull request by TheCedarPrince**
Hi folks,

Here is a possible docs PR to the Unit Tests section that explains an example workflow for creating tests for one's own package. This was inspired by my own difficulties sometimes with creating tests as well as discussion on [this Discourse post about lack of information on how to create tests](https://discourse.julialang.org/t/choosing-a-numerical-programming-language-for-economic-research-julia/85697/8?u=thecedarprince) . This was a possible idea and it may be better suited for Pkg.jl's docs but I was torn as I thought for discoverability and beginners, this may be a better location.

Either way, feel free to reject, modify, accept, or request revisions.

Thanks and have an awesome day!

### CODE REVIEW

File: stdlib/Test/docs/src/index.md [hunk 1]
```diff
@@ -332,6 +332,148 @@ Test.detect_ambiguities
...
+It is standard practice to create a file within the `test` directory called `runtests.jl` which contains the test sets we want to run.
+Go ahead and create that file within the `test` directory and add the following code to it:
+
+```julia
+using Example
+using Test
+
+@testset "Example tests" begin 
+
+	@testset "Math tests" begin
```
Comment (Seelengrab) [inline stdlib/Test/docs/src/index.md:396]: You can leave the `begin` out on master here:

```
shell> cat test.jl
foo() = @test true

@testset foo()

julia> using Test

julia> @testset "My Tests" include("test.jl")
Test Summary: | Pass  Total  Time
My Tests      |    1      1  0.3s
Test.DefaultTestSet("My Tests", Any[Test.DefaultTestSet("foo", Any[], 1, false, false, true, 1.660635314428429e9, 1.660635314582263e9, false)], 0, false, false, true, 1.660635314317867e9, 1.660635314582275e9, false)
```

Comment (TheCedarPrince) [reply stdlib/Test/docs/src/index.md:396]: Good point. As 1.8 has just been released, I think to afford backwards compatibility, I may leave in the `begin` syntax. Does that seem reasonable to you @seelengrab?

Comment (TheCedarPrince) [reply stdlib/Test/docs/src/index.md:396]: Good point. As 1.8 has just been released, I think to afford backwards compatibility, I may leave in the `begin` syntax. Does that seem reasonable to you @seelengrab?

File: stdlib/Test/docs/src/index.md [hunk 2]
```diff
@@ -332,6 +332,148 @@ Test.detect_ambiguities
...
+	@test 2 == simple_add(1, 1)
+	@test 3.5 == simple_add(1, 2.5)
+        @test_throws MethodError simple_add(1, "A")
+        @test_throws MethodError simple_add(1, 2, 3)
+end
+
+@testset "Testset 2" begin
+	@test 1.0 == type_multiply(1.0, 1.0)
+        @test isa(type_multiply(2.0, 2.0), Float64)
+	@test_throws MethodError type_multiply(1, 2.5)
```
[the remark this answers is not in the document]
Comment (TheCedarPrince) [reply stdlib/Test/docs/src/index.md:426]: Good suggestion. Will implement.

**Comment by TheCedarPrince** [comment]
From slack from mcabbott Random comments: this seems nice to have. One thing I’d like to add is mention of what files now exist. e.g. after generate Example you have src, Project.toml, etc. And after activate .   (test) pkg> add Test you now have test/Project.toml.

File: stdlib/Test/docs/src/index.md (hunk 1 above)
Comment (Seelengrab) [reply stdlib/Test/docs/src/index.md:396]: The docs probably won't be backported, so there shouldn't be any harm in changing it - I'm ambivalent either way :shrug: The idea was to have an example using that pattern in the docs and not just the docstring, to show people that they can group their tests like that much more easily.

**Comment by ViralBShah** [comment]
Just thinking out aloud, would it make sense to include this in the Pkg docs?

**Comment by ViralBShah** [comment]
Let's get this merged. @TheCedarPrince Do you have other improvements you would like to add?

**Pull request merged**